### PR TITLE
OCPBUGS-21343: [release-4.12] Bump golang.org/x/net to v0.17.0

### PR DESCRIPTION
**Pull request by awgreene**

### CODE REVIEW

File: vendor/github.com/go-errors/errors/cover.out
Comment (ncdc) [inline vendor/github.com/go-errors/errors/cover.out:1]: Remove again 😄

Comment (awgreene) [reply vendor/github.com/go-errors/errors/cover.out:1]: Blargh, I wish I was sharper.

**Comment by awgreene** [comment]
/retest
Failed cluster creation

**Comment by awgreene** [comment]
> docker build -f Dockerfile -t quay.io/operator-framework/rukpak:latest bin/linux
make: docker: Command not found
make: *** [build-container] Error 127
{"component":"entrypoint","error":"wrapped process failed: exit status 2","file":"k8s.io/test-infra/prow/entrypoint/run.go:84","func":"k8s.io/test-infra/prow/entrypoint.Options.internalRun","level":"error","msg":"Error executing test process","severity":"error","time":"2023-10-17T18:05:43Z"}
error: failed to execute wrapped command: exit status 2 

That's a new one. Will need to look into this tonight.

**Comment by ncdc** [comment]
@awgreene you can ignore the failure. In the release-4.13 branch, this job "succeeds" because the e2e make target is a no-op. We don't run this in 4.14+.

**Review by ncdc — APPROVED** [review]

**Comment by ncdc** [comment]
/hold
Want to run a payload test because we don't have proper e2e configured here

**Comment by ncdc** [comment]
@awgreene please remove the envtest commit and rebase.

Will wait for https://github.com/openshift/release/pull/44540 to merge and the configs to be updated so we can run full e2e

**Comment by ncdc** [comment]
Ok, the o/release PR is in 😄

**Pull request merged**